### PR TITLE
XAS_TDP| clean-up + performance upgrade

### PR DESCRIPTION
**Pull request by abussy**
Latest XAS_TDP code updates, including some cleaning-up and performance enhancing changes.

### CODE REVIEW

File: src/hfx_load_balance_methods.F [hunk 1]
```diff
@@ -43,7 +43,7 @@ MODULE hfx_load_balance_methods
    PRIVATE
 
    PUBLIC :: hfx_load_balance, &
-             hfx_update_load_balance, &
+             hfx_update_load_balance, reshuffle, &
```
Comment (dev-zero) [inline src/hfx_load_balance_methods.F:46]: maybe rename the procedure to something a little less generic?

File: src/xas_tdp_atom.F [hunk 2]
```diff
@@ -3206,17 +3195,13 @@ END SUBROUTINE integrate_so_prod
 !> \param intso the integral in terms of the spherical orbitals (well, their derivative)
 !> \param V the potential (put on the grid and weighted) to integrate
 !> \param ikind the atomic kind for which we integrate
-!> \param xas_atom_env ...
 !> \param qs_env ...
-!> \note The atomic grids are taken fron xas_atom_env and the orbitals are the normal ones. Ok since
```
Comment (dev-zero) [inline src/xas_tdp_atom.F:3211]: "fron" => "from"

File: src/hfx_load_balance_methods.F (hunk 1 above)
Comment (dev-zero) [reply src/hfx_load_balance_methods.F:46]: you can also wait with that, going to replace the whole thing with #1171

Comment (abussy) [reply src/hfx_load_balance_methods.F:46]: That would be a nice general tool to have, indeed.

File: src/xas_tdp_atom.F (hunk 2 above)
Comment (abussy) [reply src/xas_tdp_atom.F:3211]: Well, this goes away anyway. I am sure that if you look hard enough, you might find a few more typos in my comments

Comment (dev-zero) [reply src/xas_tdp_atom.F:3211]: ufff, right, my bad

**Pull request merged**